### PR TITLE
Fixes for 1.0

### PR DESCRIPTION
**Pull request by ranjanan**

### CODE REVIEW

File: src/Pidfile.jl [hunk 1]
```diff
@@ -202,13 +202,13 @@ function open_exclusive(path::String;
...
         catch ex
-            isa(ex, UVError) || rethrow(ex)
+            isa(ex, IOError) || rethrow(ex)
             sleep(poll_interval) # if the watch failed, convert to just doing a sleep
         end
         # now try again to create it
         file = tryopen_exclusive(path, mode)
         file === nothing || return file
-        wait(t) # sleep for a bit before trying again
+        fetch(t) # sleep for a bit before trying again
```
Comment (vtjnash) [inline src/Pidfile.jl]: ignore the deprecation – this should stay as `wait`

File: test/runtests.jl [hunk 2]
```diff
@@ -135,7 +135,7 @@ end
         close(f2)
     end
     rm("pidfile")
-    wait(rmtask)
+    fetch(rmtask)
```
Comment (vtjnash) [inline test/runtests.jl]: same here

File: test/runtests.jl [hunk 3]
```diff
@@ -165,7 +165,7 @@ end
...
```
Comment (vtjnash) [inline test/runtests.jl]: and here

File: .travis.yml [hunk 4]
```diff
@@ -4,8 +4,7 @@ os:
   - linux
   - osx
 julia:
-  - 0.6
-  - nightly
+  - 1.0
```
Comment (vtjnash) [inline .travis.yml]: Thanks!

Other changes needed:
- [ ] remove `git:   depth: 99999999`
- [ ] add `using Pkg` to after_success scripts
- [ ] delete `cd(Pkg.dir("Pidfile"));`

**Comment by vtjnash** [comment]
For AV, replace the `.appveyor.yml` file with https://github.com/vtjnash/Glob.jl/blob/e84c7e3c3e03fd89e28c5141ff82ffb9c31038df/.appveyor.yml

**Comment by vtjnash** [comment]
I'll squash and merge with this commit message:
```
* Replace UVError with IOError
* Fix other depwarns
* Update travis and Appveyor scripts
```

I think we need to update the `REQUIRES` file too? Then I can squash, merge, tag, and release – CI results look successful.

File: REQUIRE [hunk 5]
```diff
@@ -1,2 +1,2 @@
-julia v0.6
+julia 1.0
```
[the remark this answers is not in the document]
Comment (ranjanan) [reply REQUIRE]: Yeah, typo. I just changed the REQUIRE file to be consistent.

**Comment by DrDaleks** [comment]
Interestingly, when doing this same changes on my side, Pkg.test("Pidfile") was failing on my side:
```
Test Summary:                  | Pass  Fail  Error  Total
Pidfile.jl                     |   65     1      1     67
  validpid                     |    9                   9
  write_pidfile                |   14                  14
  open_exclusive               |   27                  27
  open_exclusive: break lock   |    4                   4
  open_exclusive: other errors |          1             1
  mkpidlock                    |   11            1     12
```
I'm assuming yours worked? If so, what could have happened on my side? (macOS)

**Comment by rdeits** [comment]
Any chance we could keep this working on v0.7? Otherwise there's no deprecation-free version of Pidfile.jl available while we're transitioning through 0.7.

**Comment by ranjanan** [comment]
@vtjnash I think this is good to merge. Should we keep the REQUIRE at 0.7 instead of 1.0? What do you prefer?

**Comment by ranjanan** [comment]
I actually think v0.7 is a better idea so that other packages don't get blocked

**Comment by ranjanan** [comment]
@vtjnash if you think this PR looks fine, can we merge it?

File: REQUIRE [hunk 6]
```diff
@@ -1,2 +1,2 @@
-julia v0.6
-Compat v0.30.0
+julia 0.7
+Compat 0.30.0
```
Comment (vtjnash) [inline REQUIRE]: oops, I should have just had you delete Compat and all uses of it (since it's now not actually used). Do you mind?

**Pull request merged**